java/ql/src/experimental/Security/CWE/CWE-352/JsonpInjectionLib.qll

```diff
@@ -0,0 +1,130 @@
+import java
+import DataFlow
+import JsonStringLib
+import semmle.code.java.security.XSS
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.frameworks.spring.SpringController
+
+/** Taint-tracking configuration tracing flow from user-controllable function name jsonp data to output jsonp data. */
+class VerificationMethodFlowConfig extends TaintTracking::Configuration {
+  VerificationMethodFlowConfig() { this = "VerificationMethodFlowConfig" }
+
+  override predicate isSource(DataFlow::Node src) { src instanceof RemoteFlowSource }
+
+  override predicate isSink(DataFlow::Node sink) {
+    exists(MethodAccess ma, BarrierGuard bg |
```

Comment:
Minor: The tests are not covering `BarrierGuard`. I can remove `BarrierGuard bg` from this `exists` and the tests still pass.

Reply:
@tamasvajk I don't understand this answer too much, I hope to give a test example, thank you.

Reply:
I removed the `BarrierGuard` altogether from this predicate, and checked if this change made any difference in the expected test files. It didn't, because I think all the verification methods in the tests are all correctly controlling an `if`. You could add an erroneous "verification" method access too, which doesn't guard any branches. Then the `BarrierGuard` could not be removed (accidentally in a refactoring for example) without a test picking up the change.